docs/common/format/spec.md

```diff
@@ -549,7 +551,10 @@ For example, an `events` table with a timestamp column named `ts` that is partit
 
 Scan predicates are also used to filter data and delete files using column bounds and counts that are stored by field id in manifests. The same filter logic can be used for both data and delete files because both store metrics of the rows either inserted or deleted. If metrics show that a delete file has no rows that match a scan predicate, it may be ignored just as a data file would be ignored [2].
 
-Data files that match the query filter must be read by the scan.
+Data files that match the query filter must be read by the scan. Note that for any snapshot, all file paths
+may appear at most once across all manifest files in the snapshot. If a file path appears more then once, the results of the scan are undefined. Reader implementations may raise an error in this case, but are
```

Comment:
I think that the readers may raise an error or may read the file twice. I'm not sure that I'd consider the result undefined, although I see what you're saying. If the scan doesn't check that data files are unique, then the result should be the file is read twice. If it does, then it can fail with an error.

Reply:
> I'm not sure that I'd consider the result undefined, although I see what you're saying.

I was using undefined behavior in the [PL sense](https://en.wikipedia.org/wiki/Undefined_behavior#:~:text=In%20computer%20programming%2C%20undefined%20behavior,which%20the%20computer%20code%20adheres.).  To be pedantic, I don't think here is statement in the spec yet, about the DELETE being informational only, so that adds to the possibilities, but I didn't list them.

Reply:
I stand corrected the first paragraph of this section mentions deleted files, I think I maybe missed it because I was searching for "DELETE" status.

Reply:
I'm wondering if it's worth to add some checks in the core (e.g. introducing `FastAppend::validate` and `MergeAppend::validate`) to check if data files were appended more than once.